pkg/factory/http.go

```diff
@@ -163,6 +166,51 @@ func customHTTPClient(f *Factory) func() (*http.Client, error) {
 	}
 }
 
+type basicAuthTransport struct {
+	username, password, useragent, accept string
+	underlyingTransport                   http.RoundTripper
+}
+
+func (bat basicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Add("User-Agent", bat.useragent)
+	req.Header.Add("Accept", bat.accept)
+	req.SetBasicAuth(bat.username, bat.password)
+
+	// overwrite Accept header if we have anything in the context
+	if accept, ok := req.Context().Value(ContextAcceptValue).(string); ok {
+		req.Header.Set("Accept", accept)
+	}
+
+	return bat.underlyingTransport.RoundTrip(req)
+}
+
+func basicAuthClient(f *Factory) func(username, password string) (*http.Client, error) {
+	return func(username, password string) (*http.Client, error) {
+		client, err := f.HTTPClient()
+		if err != nil {
+			return nil, err
+		}
+		parentTransport, err := f.HTTPTransport()
+		if err != nil {
+			return nil, err
+		}
+		if u := os.Getenv("SDPCTL_DOCKER_REGISTRY_USERNAME"); len(u) > 0 {
+			username = u
+		}
+		if p := os.Getenv("SDPCTL_DOCKER_REGISTRY_PASSWORD"); len(p) > 0 {
+			password = p
+		}
+		client.Transport = basicAuthTransport{
+			underlyingTransport: parentTransport,
+			username:            username,
+			password:            password,
+			useragent:           f.userAgent,
+			accept:              fmt.Sprintf("application/vnd.appgate.peer-v%d+json", f.Config.Version),
+		}
+		return client, nil
+	}
+}
+
```

Comment:
Why is this needed? wouldn't it be enough to simply add the Authorization header key, value on the request?

Reply:
May be a bit overworked here, but it's mainly to make the check for authorization a bit cleaner in the prepare command, but now that I look at it, there's a bit of double checking the env variables too, so will fix that. Anyway, there's quite a lot of requests being made against the registry, so not having to deal with setting a header at each request was a conveniance, but I can simplify it if that's preferred.